subsys/random/Kconfig

```diff
@@ -17,6 +17,7 @@ config TEST_RANDOM_GENERATOR
 
 choice RNG_GENERATOR_CHOICE
 	prompt "Random generator"
+	default XOSHIRO_RANDOM_GENERATOR if RNG_GENERATOR_DEFAULT_XOSHIRO && ENTROPY_HAS_DRIVER
```

Comment:
> Kconfig does not allow to select an option which is a
> choice. Additional Kconfig option is added to allow that.

using a `default` for this probably won't work because a choice has a prompt, which means that if config in the choice is already active, then a selecting a symbol which then selects `RNG_GENERATOR_DEFAULT_XOSHIRO` which then changes the default choice will not have an effect because of the stuck symbol syndrome.

Ref: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#stuck-symbols-in-menuconfig-and-guiconfig

Reply:
applied changes from upstream PR

Reply:
@tejlmand can you take another look?

Reply:
@tejlmand ping